recipe/run_test.py

```diff
@@ -14,18 +14,20 @@
 import pkgutil
 pkgutil.find_loader("cupy")
```

Comment:
Do you need the lines 5-15? Those are checked in line 27 right?

Reply:
Lines 5-8 need to be kept as a check for our activation/deactivation scripts. The rest can be removed 👍

Reply:
I looked at the activate script and hijacking `CUDA_PATH` env variable like this is wrong IMO. Usually people expect nvcc under `$CUDA_PATH/bin/nvcc` and even if they set it for another software, when cupy env is activated the env variable is hijacked.
Maybe cupy should use `CUPY_CUDA_PATH` and fallback to `CUDA_PATH`.

Reply:
This is admittedly a hack, but is a delicate one after a long investigation, see the linked issues in #49. Sorry that the discussions were kinda scattered.

> Usually people expect nvcc under `$CUDA_PATH/bin/nvcc`

No. The logic is always* reversed: if `CUDA_PATH` is not set, then it's looked up by finding where `nvcc` is (using `which`). 

(* including CuPy, PyTorch, RMM, and Numba)

> even if they set it for another software, when cupy env is activated the env variable is hijacked.

For all conda software, they'll see the same `cudatoolkit` with the hijacked `CUDA_PATH`.

For all non-conda software, hijacking `CUDA_PATH` is also the right thing to do if dynamic linking is used, so that all software (conda or not) see the same `cudatoolkit`. If "static" linking is used, then it doesn't matter.

Moreover, on Linux systems the standard env var to set is `CUDA_HOME`, not `CUDA_PATH`. But we only provide Linux support in this feedstock, so the chances for collision is very low.

> Maybe cupy should use `CUPY_CUDA_PATH` and fallback to `CUDA_PATH`.

Please feel free to jump in cupy/cupy#3222 (one of the issues linked in #49) 🙂